### PR TITLE
CPU Arch

### PR DESCRIPTION
**Pull request by ravipwaghmare**
Added CPU Arch Implementation under CPU Parameter
Also added test cases for the same.

This PR is related to the below proposal and issue:
https://github.com/metal3-io/metal3-docs/pull/147
https://github.com/metal3-io/hardware-classification-controller/issues/56

@dhellmann @maelk @digambar15 @russellb @zaneb 
kindly review.

### CODE REVIEW

**Comment by fmuyassarov** [comment]
@ravipwaghmare Would you mind to add a reference in the PR description to the proposal?

**Comment by Ashughorla** [comment]
@fmuyassarov Proposal link and Issue link updated in description.

File: classifier/cpu.go [hunk 1]
```diff
@@ -46,9 +47,35 @@ func checkCPU(profile *hwcc.HardwareClassification, host *bmh.BareMetalHost) boo
 		return false
 	}
 
+	ok = checkArch(
```
Comment (digambar15) [inline classifier/cpu.go]: I suggest to change the method name checkArch to checkCPUArch

**Review by digambar15 — CHANGES_REQUESTED** [review]
Minor changes required. Rest looks ok.

File: classifier/cpu.go (hunk 1 above)
Comment (ravipwaghmare) [reply classifier/cpu.go]: Done

**Review by digambar15 — APPROVED** [review]
Looks good!

**Comment by digambar15** [comment]
Thanks Ravi!

**Pull request merged**